engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ReloadFileCmd.scala

```diff
@@ -0,0 +1,42 @@
+package org.enso.interpreter.instrument.command
+
+import org.enso.interpreter.instrument.execution.RuntimeContext
+import org.enso.interpreter.instrument.job.{EnsureCompiledJob, ExecuteJob}
+import org.enso.polyglot.runtime.Runtime.Api
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/** A command that performs reloading the contents of the file.
+  *
+  * @param request a request for reload
+  */
+class ReloadFileCmd(request: Api.ReloadFileNotification) extends Command(None) {
+
+  override def execute(implicit
+    ctx: RuntimeContext,
+    ec: ExecutionContext
+  ): Future[Unit] = {
+    ctx.locking.acquireFileLock(request.path)
+    try {
+      ctx.jobControlPlane.abortAllJobs()
+      ctx.jobProcessor.run(
+        new EnsureCompiledJob(Seq(request.path), forceCompilation = true)
```

Comment:
If you send module sources like `OpenFileCmd` does, and then use `ctx.executionService.setModuleSources` (which resets the compilation stage) then you don't need to change the compilation step adding `forceCompilation` flag.

Reply:
applied